src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -713,11 +713,23 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
 
             if (!result.Succeeded)
             {
-                // If the arguments had an error reported about them then suppress further error
-                // reporting for overload resolution. 
-
-                if (!analyzedArguments.HasErrors)
+                if (analyzedArguments.HasErrors)
+                {
+                    // Errors for arguments have already been reported, except for unbound lambdas.
```

Comment:
To your first question: We cannot report errors for unbound lambdas where we report errors for other arguments, as we need the unbound lambdas in order to do overload resolution.

To your second question: Yes, it *would* be a good idea to preserve these for the `SemanticModel`. I'll revise.